src/AppInstallerRepositoryCore/Rest/Schema/1_1/RestInterface_1_1.cpp

```diff
@@ -0,0 +1,156 @@
+        std::map<std::string_view, std::string> result = params;
+
+        for (auto const& param : m_information.RequiredQueryParameters)
+        {
+            if (params.end() == std::find_if(params.begin(), params.end(), [&](const auto& pair) { return Utility::CaseInsensitiveEquals(pair.first, param); }))
+            {
+                if (Utility::CaseInsensitiveEquals(param, MarketQueryParam))
+                {
+                    result.emplace(MarketQueryParam, Runtime::GetOSRegion());
+                    continue;
+                }
+
+                AICLI_LOG(Repo, Error, << "Search request is not supported by the rest source. Required query Parameter: " << param);
+                throw UnsupportedRequestException({}, {}, {}, m_information.RequiredQueryParameters);
+            }
+        }
+
+        for (auto const& param : m_information.UnsupportedQueryParameters)
+        {
+            if (params.end() != std::find_if(params.begin(), params.end(), [&](const auto& pair) { return Utility::CaseInsensitiveEquals(pair.first, param); }))
+            {
+                AICLI_LOG(Repo, Error, << "Search request is not supported by the rest source. Unsupported query Parameter: " << param);
+                throw UnsupportedRequestException({}, {}, m_information.UnsupportedQueryParameters, {});
+            }
+        }
+
+        return result;
+    }
+
+    web::json::value Interface::GetValidatedSearchBody(const SearchRequest& searchRequest) const
+    {
+        SearchRequest resultSearchRequest = searchRequest;
+
+        for (auto const& field : m_information.RequiredPackageMatchFields)
+        {
+            PackageMatchField matchField = StringToPackageMatchField(field);
+
+            if (searchRequest.Filters.end() == std::find_if(searchRequest.Filters.begin(), searchRequest.Filters.end(), [&](const PackageMatchFilter& filter) { return filter.Field == matchField; }))
+            {
+                if (matchField == PackageMatchField::Market)
+                {
+                    resultSearchRequest.Filters.emplace_back(PackageMatchFilter(PackageMatchField::Market, MatchType::CaseInsensitive, Runtime::GetOSRegion()));
+                    continue;
+                }
+
+                AICLI_LOG(Repo, Error, << "Search request is not supported by the rest source. Required package match field: " << field);
+                throw UnsupportedRequestException({}, m_information.RequiredPackageMatchFields, {}, {});
+            }
+        }
+
+        for (auto const& field : m_information.UnsupportedPackageMatchFields)
+        {
+            PackageMatchField matchField = StringToPackageMatchField(field);
+
+            if (matchField == PackageMatchField::Unknown)
+            {
+                continue;
+            }
+
+            if (searchRequest.Filters.end() != std::find_if(searchRequest.Filters.begin(), searchRequest.Filters.end(), [&](const PackageMatchFilter& filter) { return filter.Field == matchField; }))

```

Comment:
>searchRequest.Filters [](http://example.com/codeflow?start=16&length=21)

For searchRequest inclusions, should we modify the search request to not have the unsupported package Match fields?